src/sentry/workflow_engine/handlers/detector/stateful.py

```diff
@@ -269,6 +271,135 @@ def counter_names(self) -> list[str]:
+        """
+        if not postfix:
+            return f"{self.detector.id}"
+        return f"{self.detector.id}:{postfix}"
+
+    def create_resolve_message(self) -> StatusChangeMessage:
+        """
+        Create a resolve message for the detectors issue. This is overridable in the subclass, but
+        will work for the majority of cases.
+        """
+        return StatusChangeMessage(
+            fingerprint=self.build_fingerprint(),
+            project_id=self.detector.project_id,
+            new_status=GroupStatus.RESOLVED,
+            new_substatus=None,
+        )
+
+    def _create_decorated_issue_occurrence(
+        self,
+        detector_occurrence: DetectorOccurrence,
+        evaluation_result: ProcessedDataConditionGroup,
+        new_priority: DetectorPriorityLevel,
+    ) -> IssueOccurrence:
+        """
+        Decorate the issue occurrence with the data from the detector's evaluation result.
+        """
+        evidence_data = {
+            **detector_occurrence.evidence_data,
+            "detector_id": self.detector.id,
+            "value": new_priority,
+            "conditions": [
+                result.condition.get_snapshot() for result in evaluation_result.condition_results
+            ],
+        }
+
+        return detector_occurrence.to_issue_occurrence(
+            occurrence_id=str(uuid4()),
+            project_id=self.detector.project_id,
+            status=new_priority,
+            detection_time=datetime.now(UTC),
+            additional_evidence_data=evidence_data,
+            fingerprint=[self.build_fingerprint()],
+        )
+
+    def _evaluation_detector_conditions(
+        self, value: DataPacketEvaluationType
+    ) -> tuple[ProcessedDataConditionGroup | None, DetectorPriorityLevel]:
+        """
+        Evaluate the detector.workflow_condition_group against the value in the data packet.
+
+        Returns a tuple of the condition evaluation and the new priority level.
+        """
+        new_priority = DetectorPriorityLevel.OK
+        if not self.condition_group:
+            metrics.incr("workflow_engine.detector.skipping_invalid_condition_group")
+            return None, new_priority
+
+        condition_evaluation, _ = process_data_condition_group(self.condition_group.id, value)
+
+        if condition_evaluation.logic_result:
+            validated_condition_results: list[DetectorPriorityLevel] = [
+                condition_result.result
+                for condition_result in condition_evaluation.condition_results
+                if condition_result.result is not None
+                and isinstance(condition_result.result, DetectorPriorityLevel)
+            ]
+
+            new_priority = max(new_priority, *validated_condition_results)
+
+        return condition_evaluation, new_priority
+
+    def evaluate(
+        self, data_packet: DataPacket[DataPacketType]
+    ) -> dict[DetectorGroupKey, DetectorEvaluationResult] | None:
```

Comment:
fyi - planning to update this result to be: `DetectorEvaluationResult |
 dict[DetectorGroupKey, DetectorEvaluationResult] | None` to remove the `{ None: ... }` bit in the return response, want to have that be an isolated refactor since there's a lot of things to update.